genconvert/src/main/java/com/google/myanmartools/CompileTranslit.java

```diff
@@ -0,0 +1,463 @@
+      jsOutput.add(" * you may not use this file except in compliance with the License.\n");
+      jsOutput.add(" * You may obtain a copy of the License at\n");
+      jsOutput.add(" *\n");
+      jsOutput.add(" *    http://www.apache.org/licenses/LICENSE-2.0\n");
+      jsOutput.add(" *\n");
+      jsOutput.add(" * Unless required by applicable law or agreed to in writing, software\n");
+      jsOutput.add(" * distributed under the License is distributed on an \"AS IS\" BASIS,\n");
+      jsOutput.add(" * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n");
+      jsOutput.add(" * See the License for the specific language governing permissions and\n");
+      jsOutput.add(" * limitations under the License.\n");
+      jsOutput.add(" */\n");
+      jsOutput.add("\n");
+      jsOutput.add("package com.google.myanmartools;\n");
+      jsOutput.add("\n");
+      jsOutput.add("/**\n");
+      jsOutput.add(" * Transliteration initialization of phases and rules.\n");
+      jsOutput.add(" * Note: this is generated by compiling ICU transliteration rule format.\n");
+      jsOutput.add(" * Source rule file = " + inputFilename + "\n");
+      jsOutput.add(" */\n");
+      jsOutput.add("\n");
+      jsOutput.add("import com.google.myanmartools.Transliterate;\n");
+      jsOutput.add("import com.google.myanmartools.Phase;\n");
+      jsOutput.add("import com.google.myanmartools.Rule;\n");
+      jsOutput.add("\n");
+
+      String className = "Transliterate" + nameSuffix;
+      jsOutput.add("public class " + className + " extends Transliterate {\n");
+
+      jsOutput.add("  public " + className + "() {\n");
+
+      return jsOutput;
+    }
+
+    public ArrayList<String> generateJava(String inputFilename, String nameSuffix) {
+      //For each phase in all phases.
+
+      ArrayList<String> out = javaBasicStuff(inputFilename, nameSuffix);
+
+      int phaseNum = 0;
+      for (TranslitPhase phase: phases) {
+        String phaseName = "phase" + phaseNum;
+        out.add("\n    // Rules for phase " + phaseNum + "\n");
+        out.add("    Phase " + phaseName + " = addPhase();\n");
+
+        for (TranslitRule rule: phase.phaseRules) {
+          // phase0.addRule(new Rule("patternA", "substA"));
+          if (rule.rulePattern != null && !"\n".equals(rule.rulePattern.substring(0, 1))) {
+            out.add("    " +  phaseName + ".addRule(new Rule(\"");
+            out.add(rule.rulePattern);
+            out.add("\", \"");
+            out.add(rule.ruleResult);
+            out.add("\")");
+
+            if (rule.atStart) {
+              out.add("\n          .setMatchOnStart()");
+            }
+            if (rule.revisitPosition >= 0) {
+              out.add("\n          .setRevisitPosition(" + rule.revisitPosition + ")");
+            }
+            if (rule.beforeContext.length() > 0) {
+              out.add("\n          .setBeforeContext(" + rule.beforeContext + ")");
```

Comment:
Doesn't the beforeContext/afterContext need to be quoted in the Java output?

It seems like this would have made the Java output not compile.  Are you sure this is being used?

Reply:
Yes, you are correct. The contexts are now quoted.